test/integration/controller/core/workload_controller_test.go

```diff
@@ -209,4 +215,158 @@ var _ = ginkgo.Describe("Workload controller", func() {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &updatedQueueWorkload)).To(gomega.Succeed())
+				return updatedQueueWorkload.Spec.Priority
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(&updatedPriority))
+		})
+		ginkgo.It("the PriorityClassSource is mutable (from workloadPriorityClass to k8sPriorityClass)", func() {
+			wl = testing.MakeWorkload("wl", ns.Name).Queue("lq").Request(corev1.ResourceCPU, "1").
+				PriorityClass("high-priority-class").PriorityClassSource(constants.WorkloadPriorityClassSource).Priority(200).Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+			gomega.Eventually(func() []metav1.Condition {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &updatedQueueWorkload)).To(gomega.Succeed())
+				return updatedQueueWorkload.Status.Conditions
+			}, util.Timeout, util.Interval).ShouldNot(gomega.BeNil())
+			initialPriority := int32(200)
+			gomega.Expect(updatedQueueWorkload.Spec.Priority).To(gomega.BeComparableTo(&initialPriority))
+
+			updatedQueueWorkload.Spec.PriorityClassSource = constants.K8sPriorityClassSource
+			updatedPriority := int32(100)
+			gomega.Expect(k8sClient.Update(ctx, &updatedQueueWorkload)).To(gomega.Succeed())
+			gomega.Eventually(func() *int32 {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &updatedQueueWorkload)).To(gomega.Succeed())
+				return updatedQueueWorkload.Spec.Priority
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(&updatedPriority))
+		})
+		ginkgo.It("the PriorityClassSource and priorityClass name are mutable", func() {
+			wl = testing.MakeWorkload("wl", ns.Name).Queue("lq").Request(corev1.ResourceCPU, "1").
+				PriorityClass("high-priority-class").PriorityClassSource(constants.WorkloadPriorityClassSource).Priority(200).Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+			gomega.Eventually(func() []metav1.Condition {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &updatedQueueWorkload)).To(gomega.Succeed())
+				return updatedQueueWorkload.Status.Conditions
+			}, util.Timeout, util.Interval).ShouldNot(gomega.BeNil())
+			initialPriority := int32(200)
+			gomega.Expect(updatedQueueWorkload.Spec.Priority).To(gomega.BeComparableTo(&initialPriority))
+
+			updatedQueueWorkload.Spec.PriorityClassSource = constants.K8sPriorityClassSource
+			updatedQueueWorkload.Spec.PriorityClassName = "low-priority-class"
+			updatedPriority := int32(10)
+			gomega.Expect(k8sClient.Update(ctx, &updatedQueueWorkload)).To(gomega.Succeed())
+			gomega.Eventually(func() *int32 {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &updatedQueueWorkload)).To(gomega.Succeed())
+				return updatedQueueWorkload.Spec.Priority
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(&updatedPriority))
+		})
+	})
+
+	// We ignore priorityClass because currently priorityClass is immutable.
```

Comment:
what did you mean here?

Reply:
Ah, you mean that there is no test for a changing k8s Priority class, only for a changing WorkloadPriorityClass. Correct?

Yeah, I would not add such test, but let's keep the one for WorkloadPriorityClass.